packages/lesswrong/lib/collections/posts/custom_fields.js

```diff
@@ -379,6 +379,88 @@ addFieldsDict(Posts, {
     hidden: false,
     control: "text"
   },
+  
+  // The next post. If a sequenceId is provided, that sequence must contain this
+  // post, and this returns the next post after this one in that sequence. If
+  // no sequenceId is provided, uses this post's canonical sequence.
+  nextPost: resolverOnlyField({
+    type: "Post",
+    graphQLtype: "Post",
+    viewableBy: ['guests'],
+    graphqlArguments: 'sequenceId: String',
+    resolver: async (post, { sequenceId }, { currentUser, Posts, Sequences }) => {
+      if (sequenceId) {
+        const nextPostID = await Sequences.getNextPostID(sequenceId, post._id);
+        if (nextPostID) {
+          const nextPost = await Posts.loader.load(nextPostID);
+          return accessFilterSingle(currentUser, Posts, nextPost);
+        }
+      }
+      if (post.canonicalNextPostSlug) {
+        const nextPost = await Posts.findOne({ slug: post.canonicalNextPostSlug });
+        return accessFilterSingle(currentUser, Posts, nextPost);
+      } else if(post.canonicalSequenceId) {
+        const nextPostID = await Sequences.getNextPostID(post.canonicalSequenceId, post._id);
+        if (nextPostID) return null;
```

Comment:
There is an exclamation mark missing here. This likely means this case has never been tested and I expect there are other problems with it. We should likely exercise this case at least once before merging.